### ISSUE TITLE
When WF farm is created it fails on missing assembly

### ISSUE THREAD

**Issue by IlleNilsson** [microsoft/WorkflowManagerDSC#5]
Cannot validate argument on parameter 'SBClientConiguration'. Could not load file or assebly 'Microsoft.ServiceBus, Version=1.8.0.0, Culture=neutral, PublicKeyToken=<SomeTokenSymbolsHere> or one of its dependencies. The system cannot find the file specified.

**Comment by rpinxten** [comment]
I have seen this message a lot when trying to recover a failed run of WorkflowManagerFarm. Basically the only way I have found to recover if anything fails during WorkflowManagerFarm, is to:
- Run C:\Program Files\Workflow Manager\1.0\Microsoft.Workflow.Deployment.ConfigWizard.exe
- Choost Leave (if available and complete the wizard)
- Remove all WF and SB databases on the SQL instance (if available)
- uninstall Workflow Manager Client, Workflow Manager and Service Bus. 
- Removing the C:\Program Files\Workflow Manager folder
- Reboot the system. 

Then run WorkflowManagerInstall and WorkflowManagerFarm configuration again.
Also the following permissions need to be set correctly:
- the PsDscRunAsCredential must be the same as the RunAsAccount (the workflow service account)
- the workflow service account must be local admin on the machine and sysadmin on the SQL instance (during installation, afterwards you can remove these excessive permissions)

Workflow installation and configuration is a real drag I'm afraid.

**Comment by IlleNilsson** [comment]
That is exactly what i have done. I have also started installations of SB and WF from scratch.

It always ends up with the same error using the latest installation media obtained via WebPI.

**Comment by IlleNilsson** [comment]
Added screen shot of installed programs and updates, DSC output, assembly versions and WebPI installer folder (without installer images for size)

![WFM Case - Installed programs](https://user-images.githubusercontent.com/34071280/66322966-4c047b00-e923-11e9-9ac7-3445f829a33c.PNG)
![WFM Case - Installed updates](https://user-images.githubusercontent.com/34071280/66322967-4c047b00-e923-11e9-9f1a-44893321827e.PNG)
[WFM Case - PowerShell DSC output.txt](https://github.com/microsoft/WorkflowManagerDSC/files/3697998/WFM.Case.-.PowerShell.DSC.output.txt)
[WFM Case - WebPI offline installer files.zip](https://github.com/microsoft/WorkflowManagerDSC/files/3697999/WFM.Case.-.WebPI.offline.installer.files.zip)
[WFM Case - Assembly versions.txt](https://github.com/microsoft/WorkflowManagerDSC/files/3698000/WFM.Case.-.Assembly.versions.txt)

**Comment by IlleNilsson** [comment]
Forked, commited and pull request done. I hope someone takes on to it.

BR // IlleNilsson

**Comment by ykuijs** [comment]
Hi @IlleNilsson, thanks for reporting this and submitting the PR. Since you also updated some project wide files (like the psd1 manifest file), I am going to take your additions in a new PR I am working on.

One question: Do you have some information/links where you found the solution you implemented in the PR? Or was this a matter of trial and error?

**Comment by IlleNilsson** [comment]
Hi

I found a workaround for when installing via PowerShell w/o DSC.

I just incorpated that workaround in the DSC Resource module.

I can’t find the source article just right now.

We have had a case open against Microsoft to fix the actual bug but the service technician just argued that it is by design and the workaround is to install SB and WF manually via the GUI. Not a solution for DSC.

BR // Ilian
